modules/Reprepro.py

```diff
@@ -0,0 +1,93 @@
+# Deb-o-Matic - Reprepro module
+#
+# Copyright (C) 2011-2015 Luca Falavigna
+#                    2016 Niko Tyni
+#
+# Authors: Niko Tyni <ntyni@debian.org>
+#
+# Based on the Repository and Lintian modules
+#  by Luca Falavigna <dktrkranz@debian.org>
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; version 3 of the License.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301, USA.
+#
+# Generate a local reprepro repository of built packages
+
+import os
+from subprocess import call
+from logging import debug, error, info
+
+
+class DebomaticModule_Reprepro:
+
+    def __init__(self):
+        self.reprepro = '/usr/bin/reprepro'
+        self.before   = [ 'BuildCleaner' ]
+
+    def post_build(self, args):
+        self.update_repository(args)
```

Comment:
Repository module updates local repository before each build as it could be packages have been dropped already by RemovePackages

Reply:
I have the impression reprepro should do the same to be consistent with the Repository plugin.

Reply:
Reprepro doesn't use the packages in the debomatic pool/ directory, it copies them into its own repository elsewhere after the build. In our case we then use BuildCleaner to remove them from the debomatic side afterwards.

So currently RemovePackages doesn't affect the reprepro repository at all. I'm not sure if it should. What do you think?